src/toil/wdl/wdltoil.py

```diff
@@ -5500,6 +5501,29 @@ def main() -> None:
                         )
                         options.all_call_outputs = True
 
+                WDL.Lint.lint(document)
+
+                strict = options.strict
+                shown = [0]
+
+                fp = io.StringIO()
+                outline(
+                    document,
+                    0,
+                    file=fp,
+                    show_called=(document.workflow is not None),
+                    shown=shown,
+                )  # type: ignore[no-untyped-call]
+
+                if getattr(WDL.Lint, "_shellcheck_available", None) is False:
+                    logger.info("Suggestion: install shellcheck (www.shellcheck.net) to check task commands")
+
+                if shown[0]:
+                    logger.warning('Workflow lint warnings:\n%s', fp.getvalue().rstrip())
+                    if strict:
+                        logger.critical(f'Workflow did not pass linting in strict mode')
+                        sys.exit(2)
```

Comment:
We could note here that MiniWDL uses exit code 2 to indicate a linting error, so we do too.